internal/ext/unsafex/unsafex.go

```diff
@@ -17,42 +17,124 @@
 // Importing this package should be treated as equivalent to importing unsafe.
 package unsafex
 
-import "unsafe"
-
-// Size is like [unsafe.Sizeof], but it is a generic function and it returns
-// an int instead of a uintptr (Go does not have types so large they would
-// overflow an int).
-func Size[T any]() int {
-	var zero T
-	return int(unsafe.Sizeof(zero))
+import (
+	"fmt"
+	"unsafe"
+)
+
+// Int is a constraint for any integer type.
+type Int interface {
+	~int8 | ~int16 | ~int32 | ~int64 | ~int |
+		~uint8 | ~uint16 | ~uint32 | ~uint64 | ~uint |
+		~uintptr
+}
+
+// Layout is the layout of a type.
+//
+// This is a more convenient abstraction that manipulating the size and
+// alignment separately.
+type Layout struct {
+	Size, Align int
+
+	// TODO: Add something like PointerFree? It's not clear if Go even exposes
```

Comment:
What would PointerFree even do? Even if we don't have a pressing need to *use* such a method, I don't see a reason to have a TODO. This doesn't need to be a general-purpose library since it's an internal package of helpers -- we only need things that we actually need to use inside of protocompile.